packages/fast-usdc/src/clientSupport.js

```diff
@@ -0,0 +1,110 @@
+// @ts-check
+import { AmountMath } from '@agoric/ertp';
+import { assertAllDefined } from '@agoric/internal';
+
+// Reusable amount scaling similar to inter-protocol's approach
+const COSMOS_UNIT = 1_000_000n;
+const scaleDecimals = num => BigInt(num * Number(COSMOS_UNIT));
```

Comment:
Using `Number` has weird edge cases. For example, if the product inside `BigInt` isn't an integer, it throws.

Let's stick to ERTP `Amount` and `Ratio` as in:

https://github.com/Agoric/agoric-sdk/blob/f0ccde6550d7f0d22fd489205cb12a6598faa2c9/packages/fast-usdc/src/cli/lp-commands.js#L42-L46

`parseRatio` will take a `Number` or a `string` (anything that passes `assertParsableNumber`), so we could relax `amountString` likewise.

For commander-style arg parsing, we could export this one:

https://github.com/Agoric/agoric-sdk/blob/f0ccde6550d7f0d22fd489205cb12a6598faa2c9/packages/fast-usdc/src/cli/lp-commands.js#L28-L36

Reply:
One of the thing clientSupport helpers do is provide the brand. I think what you're asking for is a bigger change. Again I'm not fully opposed but I think it should be separated from this PR and cover the inter-protocol helper too.